civicase.php

```diff
@@ -588,3 +589,20 @@ function _civicase_add_case_category_case_type_entity(array &$entityTypes) {
     ];
   };
 }
+
+/**
+ * Implements hook_civicrm_alterMailParams().
+ */
+function civicase_civicrm_alterMailParams(&$params, $context) {
+  if (!empty($_GET['caseid']) && strpos($params['subject'], '[case ') !== FALSE) {
+    $case_type_category = CaseCategoryHelper::getCategoryName(intval($_GET['caseid']));
+    if (empty($case_type_category)) {
+      return;
+    }
+
+    $wordReplacements = CaseCategoryHelper::getWordReplacements($case_type_category);
+    if (!empty($wordReplacements['case'])) {
+      $params['subject'] = str_replace('[case ', '[' . $wordReplacements['case'] . ' ', $params['subject']);
+    }
+  }
```

Comment:
A couple of comments. 
(1) Let's use `CRM_Utils_Request::retrieve('caseid', 'Integer')`; to fetch the case Id. 
(2) Let's move this logic to a separate class of it's own for consistency. Once we start adding more code to this hook, it will become cluttered. 
(3) To make sure that we don't mistakenly replace any other occurrence of case in the subject, can we check the position of the returned string to target the right one.
```php
strpos($params['subject'], '[case ') === 0)
```